raspberry/bzzz/sensors/pressure_sensor.py

```diff
@@ -25,40 +27,47 @@ def _getUshort(data, index):
     # return two bytes from data as an unsigned 16-bit value
     return (data[index] << 8) + data[index + 1]
 
+class PressureSensor(threading.Thread):
+    def __init__(self, DEVICE_ADDRESS=0x77, median_filter_length=5, 
```

Comment:
Can only the median filter be used?

Reply:
I based this median filter on the old barometer code. I'm not very sure which is the right filter that can be used.

Reply:
Have a look at the implementation of `Anemometer`. There, in [line 21](https://github.com/QUB-ASL/bzzz/blob/dev/raspberry/bzzz/sensors/anemometer.py#L21), the caller specifies a `data_processor`, which can be anything. 

I think you can try to mirror the implementaiton of `Anemometer`. Your implementation could start like this:

```py
class Barometer:
    """
    Barometer sensor

    This class is used to interface the barometer
    """

    def __init__(self,
                 serial_path='???',
                 baud=115200,
                 window_length=3,
                 data_processor=MedianFilter(),
                 log_file=None,
                 max_samples=100000):
    # your code goes hhere
```

A UML of the `Barometer` class can look like this:

![image](https://github.com/QUB-ASL/bzzz/assets/125415/6da34bdf-b433-4320-9ee0-be4178ff9b3d)